omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js

```diff
@@ -29,177 +28,117 @@ export default class ObsAcrossEncountersController {
-        }).then((response) => {
-          return response.results;
-        });
-        promises.push(promisedEncounter);
+    const encounterTypes = this.config.encounterTypes ? this.config.encounterType.split(',').map(c => c.trim()) : [];
+    const legacyEncounterTypes = this.config.encounterType ? this.config.encounterType.split(',').map(c => c.trim()) : [];
+    encounterTypes.push(...legacyEncounterTypes);
+    
+    const encounterPromises = encounterTypes.map(e =>
+      this.openmrsRest.get("encounter", {
+        patient: this.config.patientUuid,
+        encounterType: e,
+        v: 'custom:(uuid,encounterDatetime,encounterType:(name,description),obs:(id,uuid,value,concept:(id,uuid,name:(display),datatype:(uuid)),groupMembers:(id,uuid,display,value,concept:(id,uuid,name:(display),datatype:(uuid))))',
+        limit: this.config.maxRecords || 4,
+        fromdate: this.widgetsCommons.maxAgeToDate(this.config.maxAge),
+        order: this.order
+      }).then(response => response.results)
+    );
+
+    this.$q.all(encounterPromises).then((encounterSets) => {
+      for (let encounterSet of encounterSets){
+        this.addToSimpleEncs(encounterSet); 
       }
-
-      this.$q.all(promises).then((dataArray) => {
-        for (var n = 0; n < dataArray.length; n++){
-          this.parseEncounters(dataArray[n]); 
-        }
-      });
-    }
-  }
-
-  getMaxRecords() {
-    if (this.config.maxRecords == '' || angular.isUndefined(this.config.maxRecords)) {
-      return 4;
-    } else {
-      return this.config.maxRecords;
-    }
+    });
   }
 
   isRetired(obs) {
-    let retired = false;
-    if (obs && angular.isDefined(obs.value) && angular.isDefined(obs.value.retired)) {
-      retired = obs.value.retired;
-    }
-    return retired;
+    return Boolean(obs && obs.value && obs.value.retired);
   }
 
-  displayObs(obs) {
-    let display = "";
-    if (obs.value != null) {
-      if (angular.isDefined(obs.value.display)) {
-        //If value is a concept
-        display = obs.value.display;
-      }
-      else if (['8d4a505e-c2cc-11de-8d13-0010c6dffd0f',
-          '8d4a591e-c2cc-11de-8d13-0010c6dffd0f',
-          '8d4a5af4-c2cc-11de-8d13-0010c6dffd0f'].indexOf(obs.concept.datatype.uuid) > -1) {
-        //If value is date, time or datetime
-        var date = this.$filter('date')(new Date(obs.value), this.config.dateFormat);
-        display = date;
-      } else {
-        display = obs.value;
-      }
+  getObsValue(obs) {
+    if (['8d4a505e-c2cc-11de-8d13-0010c6dffd0f',
+        '8d4a591e-c2cc-11de-8d13-0010c6dffd0f',
+        '8d4a5af4-c2cc-11de-8d13-0010c6dffd0f'].includes(obs.concept.datatype.uuid)) {
+        // If value is date, time or datetime
+        return this.$filter('date')(new Date(obs.value), this.config.dateFormat);
+    } else if (angular.isDefined(obs.value.display)) {
+        // If value is a concept
+        return this.config.useConceptNameForValue ? obs.value.concept.display : obs.value.display
```

Comment:
This adds support for the config value `useConceptNameForValue`.

Reply:
Read by comment below first, but I don't think this is what we want here... the idea is, "if this obs is a valueDrug, do we want to use the name of the drug, or the name of the concept associated with the drug"?  I believe "obs.value.concept.display" will result in a NPE unless this is a "valueDrug" obs.

Reply:
Oops, good catch about the NPE. But isn't `obs.value.concept.display` also present for value coded?

Reply:
I don't think so... in valueCoded, "obs.value" = Concept, and a "concept" doesn't have a concept property.

Take a look at the screenshot on the ticket, and note that the object that obs.value points to is a drug, with all the drug fields like "maxDailyDose", etc....